wallet-new/src/Cardano/Wallet/Kernel/DB/Sqlite.hs

```diff
@@ -436,75 +473,172 @@ instance Ord OrdByCreationDate where
 getTxMetas :: Sqlite.Connection
            -> Offset
            -> Limit
+           -> AccountFops
+           -> Maybe Core.Address
+           -> FilterOperation Txp.TxId
+           -> FilterOperation Core.Timestamp
            -> Maybe Sorting
-           -> IO [Kernel.TxMeta]
-getTxMetas conn (Offset offset) (Limit limit) mbSorting = do
+           -> IO ([Kernel.TxMeta], Maybe Int)
+getTxMetas conn (Offset offset) (Limit limit) accountFops mbAddress fopTxId fopTimestamp mbSorting = do
     res <- Sqlite.runDBAction $ runBeamSqlite conn $ do
-        metasWithInputs  <- nonEmpty <$> SQL.runSelectReturningList paginatedInputs
-        metasWithOutputs <- nonEmpty <$> SQL.runSelectReturningList paginatedOutputs
-        return $ liftM2 (,) metasWithInputs metasWithOutputs
+
+        -- The following 3 queries are disjointed and both fetches, respectively,
+        -- @TxMeta@ @TxInput@ and @TxOutput@.
+        -- The rationale behind doing three separate queries is that SQlite does not support
+        -- array types, neither array aggregations (https://www.sqlite.org/lang_aggfunc.html).
+        -- This in particular means the list of Inputs and Outputs for each TxId must be assembled
+        -- in memory. One workarroun is to use group_concat(see link above), but this would return
+        -- the array in a string format, separated by commas.
+
+        -- The length of meta list is bounded by 50 in each realistic senario.
+        -- So the marshalling between Haskell types - UTF8 - binary SQlite types shouldn`t
+        -- be costly.
```

Comment:
Yeah, I think this is fine. It would only be problematic if we looped over a bunch of results (haskell-side) and then called back into SQlite for each element.